src/coreclr/src/jit/codegenarm.cpp

```diff
@@ -1718,9 +1727,13 @@ void CodeGen::genProfilingLeaveCallback(unsigned helper)
 
     if (r0InUse)
     {
-        if (varTypeIsGC(compiler->info.compRetType))
+        if (varTypeIsGC(compiler->info.compRetNativeType))
+        {
+            attr = emitActualTypeSize(compiler->info.compRetNativeType);
+        }
+        else if (compiler->compMethodReturnsRetBufAddr())
         {
-            attr = emitActualTypeSize(compiler->info.compRetType);
+            attr = EA_BYREF;
```

Comment:
@AndyAyersMS Just to clarify this to myself - we need EA_BYREF for RetBufAddr in case when it points to a struct with a GC field?

Reply:
Not quite -- the struct type does not matter.

The hidden return buffer pointer always has to be treated as a byref, since there are cases like reflection invoke where the storage it points to is actually on the GC heap.

Reply:
Ah I see, thanks